go/libraries/doltcore/diff/querydiff/query_differ_test.go

```diff
@@ -262,3 +362,506 @@ func testQueryDiffer(t *testing.T, test queryDifferTest) {
+	{query: `SELECT COUNT(*) c, i AS foo FROM mytable GROUP BY 2 ORDER BY foo DESC`},
+	{query: `SELECT COUNT(*) c, i AS i FROM mytable GROUP BY 2`},
+	{query: `SELECT i AS i FROM mytable GROUP BY 1`},
+	{query: "SELECT SUM(i) + 1, i FROM mytable GROUP BY i ORDER BY i"},
+	{query: "SELECT SUM(i), i FROM mytable GROUP BY i ORDER BY 1+SUM(i) ASC"},
+	{query: "SELECT i, SUM(i) FROM mytable GROUP BY i ORDER BY SUM(i) DESC"},
+	{query: `SELECT AVG(23.222000)`},
+	{query: "SELECT substring(s, 1, 1), count(*) FROM mytable GROUP BY substring(s, 1, 1)"},
+	{query: `
+		SELECT
+			i,
+			foo
+		FROM (
+			SELECT
+				i,
+				COUNT(s) AS foo
+			FROM mytable
+			GROUP BY i
+		) AS q
+		ORDER BY foo DESC
+		`,
+	},
+	{query: "SELECT n, COUNT(n) FROM bigtable GROUP BY n HAVING COUNT(n) > 2"},
+	{query: "SELECT n, MAX(n) FROM bigtable GROUP BY n HAVING COUNT(n) > 2"},
+	{query: "SELECT substring(mytable.s, 1, 5) AS s FROM mytable INNER JOIN othertable ON (substring(mytable.s, 1, 5) = SUBSTRING(othertable.s2, 1, 5)) GROUP BY 1 HAVING s = \"secon\""},
+	{query: "SELECT s,  i FROM mytable GROUP BY i ORDER BY SUBSTRING(s, 1, 1) DESC"},
+	{query: "SELECT s, i FROM mytable GROUP BY i HAVING count(*) > 0 ORDER BY SUBSTRING(s, 1, 1) DESC"},
+	{query: `SELECT t.date_col FROM (SELECT CONVERT('2019-06-06 00:00:00', DATETIME) as date_col) t GROUP BY t.date_col`},
+	{query: "SELECT i, COUNT(i) AS `COUNT(i)` FROM (SELECT i FROM mytable) t GROUP BY i ORDER BY i, `COUNT(i)` DESC"},
+	{query: `SELECT 1 FROM mytable GROUP BY i HAVING i > 1`},
+	{query: `SELECT avg(i) FROM mytable GROUP BY i HAVING avg(i) > 1`},
+	{query: `
+		SELECT s AS s, COUNT(*) AS count,  AVG(i) AS ` + "`AVG(i)`" + `
+		FROM  (
+			SELECT * FROM mytable
+		) AS expr_qry
+		GROUP BY s
+		HAVING ((AVG(i) > 0))
+		ORDER BY count DESC
+		LIMIT 10000`,
+	},
+	{query: `
+		SELECT
+			table_schema,
+			table_name,
+			CASE
+				WHEN table_type = 'BASE TABLE' THEN
+					CASE
+						WHEN table_schema = 'mysql'
+							OR table_schema = 'performance_schema' THEN 'SYSTEM TABLE'
+						ELSE 'TABLE'
+					END
+				WHEN table_type = 'TEMPORARY' THEN 'LOCAL_TEMPORARY'
+				ELSE table_type
+			END AS TABLE_TYPE
+		FROM information_schema.tables
+		WHERE table_schema = 'mydb'
+			AND table_name = 'mytable'
+		HAVING table_type IN ('TABLE', 'VIEW')
+		ORDER BY table_type, table_schema, table_name`,
+	},
+	{query: "SELECT substring(mytable.s, 1, 5) AS s FROM mytable INNER JOIN othertable ON (substring(mytable.s, 1, 5) = SUBSTRING(othertable.s2, 1, 5)) GROUP BY 1"},
+	{query: `SELECT SUBSTRING_INDEX(mytable.s, "d", 1) AS s FROM mytable INNER JOIN othertable ON (SUBSTRING_INDEX(mytable.s, "d", 1) = SUBSTRING_INDEX(othertable.s2, "d", 1)) GROUP BY 1 HAVING s = 'secon'`},
+	{query: `SELECT SUBSTRING(s, -3, 3) AS s FROM mytable WHERE s LIKE '%d row' GROUP BY 1`},
+	{query: `SELECT SUM(i) FROM mytable`},
+}
+
+// runs a subset of SELECT queries from enginetest/queries.go as a sanity check
```

Comment:
Any reason not to run all of them? If you're just checking that they run and produce no diff, then that should be fine, right?

Another interesting test would be to create the table, commit it, then insert the rows and verify that the "to" diff contains exactly the rows expected by the normal query in the engine test.

Reply:
that's a great idea